share/completions/p4.fish

```diff
@@ -0,0 +1,726 @@
+# Fish completions for Perforce (p4) RCS
+# Based on the list of commands from
+# https://www.perforce.com/perforce/r16.1/manuals/cmdref/
+# Author: Boris Aranovich https://github.com/nomaed
+
+#########################################################
+### p4 command parsing
+#########################################################
+
+function __fish_print_p4_client_name
+    set -l p4info (p4 info -s 2> /dev/null)
+    if test -e "$p4info"
+        return
+    end
+    if string match -qr '^Client unknown'
+        return
+    end
+    string match -r '^Client name: .+' $p4info | string replace 'Client name: ' ''
+end
+
+#########################################################
+
+function __fish_print_p4_user_name
+    set -l p4info (p4 info 2> /dev/null)
+    if test -e "$p4info"
+        return
+    end
+    string match -r '^User name: .+' $p4info | string replace 'User name: ' ''
+end
+
+#########################################################
+
+function __fish_print_p4_changelists -d "Used 'detailed!' as first argument to use username@workspace prefix"
+    set -l detailed
+    if test -n "$argv"
+        and test $argv[1] = "detailed!"
+        set detailed true
+        set -e argv[1]
+    end
+
+    set -l changes (p4 changes -L $argv)
+
+    set -l result
+    for line in $changes
+        set line (string trim $line)
```

Comment:
I need to trim every line individually, since the output of `p4 changes -L` has empty lines, and others are indented.
Would using `string trim` operate on every line in the output, instead of just trimming the beginning end end of the entire buffer?